examples/python/src/hello-world.py

```diff
@@ -0,0 +1,212 @@
+                        'form': FORM,
+                    },
+                ],
+            },
+        ],
+    }
+
+
+@app.route('/ping', methods=['POST'])
+def on_ping() -> dict:
+    logging.debug('ping...')
+    return {'type': 'ok'}
+
+
+@app.route('/install', methods=['GET', 'POST'])
+def on_install() -> dict:
+    print(f'on_install called with payload , {request.args}, {request.data}', flush=True)
+    _subscribe_team_join(request.json['context'])
+    return {'type': 'ok', 'data': []}
+
+
+@app.route('/first_command', methods=['POST'])
+def on_first_command():
+    print(f'/first_command called ')
+    response_message = 'Hello! response from /first_command'
+    return {'type': 'ok', 'text': response_message}
+
+
+@app.route('/bot_joined_team', methods=['GET', 'POST'])
+def on_bot_joined_team() -> dict:
+    context = request.json['context']
+    logging.info(
+        f'bot_joined_team event received for site:{context["mattermost_site_url"]}, '
+        f'team:{context["team"]["id"]} name:{context["team"]["name"]} '
+        f'{request.args} {request.data}'
+    )
+    # Here one can subscribe to channel_joined/left events as these required team_id now to be subscribed,
+    # hence use the team_id received in the event and make a call for subscribing to channel_joined/left events.
+    # Also supply {'team_id': team_id} in the request body of the subscription
+    # {
+    #    'subject': 'bot_joined_team',
+    #    'call': {
+    #        'path': '/bot_joined_team',
+    #         'expand': {
+    #             'app': 'all',
+    #             'team': 'all'
+    #         }
+    #    },
+    #    'team_id': 'team_id'   # get this team_id when bot_joined_team event occurs
+    # }
+    return {'type': 'ok', 'data': []}
+
+
+# Subscribing to events. For example, Subscribe to 'bot_joined_team' event
+def _subscribe_team_join(context: dict) -> None:
+    site_url = context['mattermost_site_url']
+    bot_access_token = context['bot_access_token']
+    url = site_url + 'plugins/com.mattermost.apps/api/v1/subscribe'
+    logging.info(f'Subscribing to team_join for {site_url}')
+    headers = {'Authorization': f'BEARER {bot_access_token}'}
+    body = {
+        'subject': 'bot_joined_team',
+        'call': {
+            'path': '/bot_joined_team',
+            'expand': {
+                'app': 'all',
+                'team': 'all'
+            }
+        },
+    }
+    res = requests.post(url, headers=headers, json=body)
+    if res.status_code != 200:
+        logging.error(f'Could not subscribe to team_join event for {site_url}')
+    else:
+        logging.debug(f'subscribed to team_join event for {site_url}')
+
+
+if __name__ == '__main__':
+    # load settings
+    root_dir = os.path.dirname(os.path.dirname(__file__))
+    with open(f'{root_dir}/settings.json') as config_file:
+        settings = json.load(config_file)
+        for env_name, env_value in settings.items():
+            os.environ[env_name] = env_value
```

Comment:
If we do use the `settings.json` file, the environment variables should take precedence over the `settings.json` file, rather than the `settings.json` file editing the environment here. At the moment, the environment variables will not do anything unless we delete the `settings.json` file.

Reply:
Removed `settings.json` file and will use the configured environment variables. The app will use default ones in case not configured.